api/server/httputils/write_log_stream.go

```diff
@@ -17,11 +18,15 @@ import (
 // WriteLogStream writes an encoded byte stream of log messages from the
 // messages channel, multiplexing them with a stdcopy.Writer if mux is true
 func WriteLogStream(_ context.Context, w io.Writer, msgs <-chan *backend.LogMessage, config *container.LogsOptions, mux bool) {
+	// Used instead of the Flush() we had previously, so we don't set the header twice because of the otel wrapper
+	// see here: https://github.com/moby/moby/issues/47448
+	if rw, ok := w.(http.ResponseWriter); ok {
+		rw.WriteHeader(http.StatusOK)
+	}
+
 	wf := ioutils.NewWriteFlusher(w)
 	defer wf.Close()
 
-	wf.Flush()
```

Comment:
Thanks! Was wondering if we’d now be missing the “Flush” (which was the original reason), and potentially would get back to the original issue; does the WriteHeader make sure it’s sent immediately, or would it still potentially have to be flushed?

Reply:
yep, good catch. Only 1xx headers are sent immediately